engine/src/main/java/com/redhat/cloud/notifications/events/EndpointProcessor.java

```diff
@@ -96,4 +98,35 @@ public void process(Event event) {
             }
         });
     }
+
+    /**
+     * Processes the test event with the given endpoint's UUID.
+     * @param event the event to process.
+     * @param endpointUuid the endpoint's UUID.
+     */
+    public void processTestEvent(final Event event, final UUID endpointUuid) {
+        processedItems.increment();
+
+        final Endpoint endpoint = this.endpointRepository.findByUuid(endpointUuid);
+
+        DelayedThrower.throwEventually(DELAYED_EXCEPTION_MSG, accumulator -> {
+            final EndpointTypeProcessor processor;
+
+            switch (endpoint.getType()) {
+                case CAMEL:
+                    processor = camelProcessor;
+                    break;
+                case WEBHOOK:
+                    processor = webhookProcessor;
+                    break;
+                case EMAIL_SUBSCRIPTION:
+                    processor = emailProcessor;
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unexpected endpoint type: " + endpoint.getType());
+            }
+
+            processor.process(event, List.of(endpoint));
+        });
+    }
```

Comment:
Instead of doing this, we could replace
https://github.com/RedHatInsights/notifications-backend/blob/a55d5e62e4268dac31fb6ac789d47aa1e07cd2d4/engine/src/main/java/com/redhat/cloud/notifications/events/EndpointProcessor.java#L56-L58
with
```java
    private static boolean isIntegrationTest(Event event) {
        return event.getAction().getBundle().equals("console") &&
                event.getAction().getApplication().equals("notifications") &&
                event.getAction().getEventType().equals("integration-test");
    }

    public void process(Event event) {
        processedItems.increment();

        List<Endpoint> endpoints;
        if (isIntegrationTest(event)) {
            Endpoint endpoint = endpointRepository.findByUuid(event.getOrgId(), /* pass the endpoint ID from the action context */);
            endpoints = List.of(endpoint);
        } else {
            endpoints = endpointRepository.getTargetEndpoints(event.getOrgId(), event.getEventType());
        }
```